[CI] issue: HPCINFRA-3581 Clang-tools-extra 20.1.8

NirWolfer opened the pull request:
## Description
Clang version in CSbuild is old, and needs to be updated

##### What
Upgrade Clang-tools-extra from 18.1 to 19.1

##### Why ?
[HPCINFRA-3581](https://jirasw.nvidia.com/browse/HPCINFRA-3581)

##### How ?
_It is optional but for complex PRs please provide information about the design,
architecture, approach, etc._

## Change type
What kind of change does this PR introduce?
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

## Check list
- [ ] Code follows the style de facto guidelines of this project
- [ ] Comments have been inserted in hard to understand places
- [ ] Documentation has been updated (if necessary)
- [ ] Test has been added (if possible)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling in network queue operations with proper state tracking.
  * Fixed uninitialized memory access in system configuration to prevent undefined behavior.
  * Enhanced system call error handling for better reliability.

* **Refactor**
  * Simplified cache configuration validation logic.

* **Chores**
  * Updated build infrastructure and toolchain configuration.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

NirWolfer committed:
[CI] issue: HPCINFRA-3581 Clang-tools-extra 20.1.8

Upgrade Clang-tools-extra from 18.1 to 20.1.8

Signed-off-by: NirWolfer <nwolfer@nvidia.com>

NirWolfer committed:
[CI] issue: HPCINFRA-3581 Clang-tools-extra 20.1.8

Upgrade Clang-tools-extra from 18.1 to 20.1.8

Signed-off-by: NirWolfer <nwolfer@nvidia.com>

dpressle commented on .ci/dockerfiles/Dockerfile.rhel8.6:
```diff
@@ -5,24 +5,32 @@ ARG _UID=6213
+    yum makecache && \
+    yum install -y yum-utils clang-tools-extra-19.1.7 sudo curl autoconf automake make libtool \
+    libnl3-devel libnl3 rdma-core-devel rdma-core bc && \ 
+    yum-config-manager --add-repo https://archives.fedoraproject.org/pub/archive/epel/7/x86_64/ && \
```
That this mean we need epel7 for cppcheck and epel8 for csbuild ?

NirWolfer replied:
yes, double checked to make sure we are indeed installing these packages from these repos

dpressle commented on .ci/dockerfiles/Dockerfile.rhel8.6:
```diff
@@ -5,24 +5,32 @@ ARG _UID=6213
     useradd -u "$_UID" -g "$_GID" -s /bin/bash -m -d ${_HOME} "${_LOGIN}" && \
     chown -R ${_LOGIN} ${_HOME} && \
     mkdir /build && chown -R ${_LOGIN} /build
+RUN rm -f /etc/yum.repos.d/sw-rivermax ubi.repo
```
add comment here why we are removing this repo - in few weeks we will forget why we did it

dpressle reviewed (changes requested):
please check release build i want to make sure we can build release image without UBI

galnoam commented:
@tomerdbz , @pasis , do we need this PR?
What is the blocker here?

tomerdbz commented:
@galnoam @pasis See ticket: https://jirasw.nvidia.com/browse/HPCINFRA-3581
we use in csbuild an old version of the analyzers with some bugs that were fixed.
this made us add A LOT of suppressions and we might miss real issues in the current state.

tomerdbz commented:
bot:retest

tomerdbz committed:
issue: HPCINFRA-3581 Fix clang-20 compilation errors

Add missing override specifier to ring_slave::tls_rx_create_rule()
and remove redundant std::max with unsigned zero in UTLS watermark
configuration.

Signed-off-by: Tomer Cabouly <tcabouly@nvidia.com>

tomerdbz committed:
issue: HPCINFRA-3581 Fix clang-20 compilation errors

Add missing override specifier to ring_slave::tls_rx_create_rule()
and remove redundant std::max with unsigned zero in UTLS watermark
configuration.

Signed-off-by: Tomer Cabouly <tcabouly@nvidia.com>

tomerdbz committed:
issue: HPCINFRA-3581 Fix csbuild static analysis warnings

Initialize errno before reading it in hw_queue_rx drain loop,
zero-initialize ethtool_value in validate_lro to silence cppcheck,
and fix sys_sendto to only check errno after sendto returns -1.

Signed-off-by: Tomer Cabouly <tcabouly@nvidia.com>

tomerdbz committed:
issue: HPCINFRA-3581 Fix csbuild static analysis warnings

Initialize errno before reading it in hw_queue_rx drain loop,
zero-initialize ethtool_value in validate_lro to silence cppcheck,
and fix sys_sendto to only check errno after sendto returns -1.

Signed-off-by: Tomer Cabouly <tcabouly@nvidia.com>

NirWolfer commented:
As per @tomerdbz request, i changed the clang tools version to 20.1.8

pasis commented:
bot:retest

pasis commented:
The gtests failure happened likely due xlio json config misconfiguration - unrelated to current PR and needs to be fixed separately.

pasis commented on .ci/dockerfiles/Dockerfile.rhel8.6:
```diff
@@ -16,13 +17,14 @@ RUN echo "${_LOGIN} ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
- && yum clean all
+RUN yum makecache && \
+    yum install -y yum-utils clang-tools-extra-20.1.8 sudo curl autoconf automake make libtool \
+    libnl3-devel libnl3 rdma-core-devel rdma-core bc && \ 
```
fixed

pasis commented on .ci/dockerfiles/Dockerfile.rhel8.6:
```diff
@@ -16,13 +17,14 @@ RUN echo "${_LOGIN} ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
+    yum install -y yum-utils clang-tools-extra-20.1.8 sudo curl autoconf automake make libtool \
+    libnl3-devel libnl3 rdma-core-devel rdma-core bc && \ 
+    yum-config-manager --add-repo https://archives.fedoraproject.org/pub/archive/epel/7/x86_64/ && \
+    yum --nogpgcheck install -y cppcheck && \
```
needs to be checked and updated consistently across all the places as a separate task.

pasis commented on tools/daemon/daemon.c:
```diff
@@ -385,6 +385,9 @@ ssize_t sys_sendto(int sockfd, const void *buf, size_t len, int flags,
             }
             return -errno;
         }
+        if (n == 0) {
+            break;
+        }
```
daemon is going to be removed soon.
